File: Benchmarking_Languages_Project/Julia_Tests/Test16.jl
load("SA.jl")

function energyFnc(x) 
 ret = 10*2 + (x[1]^2 - 10*cos(2*pi*x[1])) + (x[2]^2 - 10*cos(2*pi*x[2]))
 return ret
end

function neighFnc(nParams,lowerBounds,upperBounds)
 Nhood = zeros(nParams)
  for i in 1:2
   Nhood[i] = randi((lowerBounds[i],upperBounds[i])) + rand()
  end
 return Nhood
end

tic()
for i in 1:15
T16_1 = runSA(2,[-5,-5],[5,5],energyFnc,neighFnc)
end
timing = toc()
println(timing/15)

File: Benchmarking_Languages_Project/Python_Tests/Test16.py
execfile("SA.py")

def energyFnc(x):
	ret = 10*2 + (x[0]**2 - 10*cos(2*pi*x[0])) + (x[1]**2 - 10*cos(2*pi*x[1]))
	return ret

def neighFnc(nParams,lowerBounds,upperBounds):
 Nhood = zeros(nParams)
 for i in range(nParams):
  Nhood[i] = uniform(lowerBounds[i],upperBounds[i]) 
 return Nhood

time1 = time()
T16 = runSA(2,[-5.1,-5.1],[5.1,5.1],energyFnc,neighFnc)
print(time() - time1)
